### PR TITLE
feat: Replace SelectResult with FlightData

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR is part of a series of changes to gradually adopt the Arrow Flight, and remove the old GRPC interface finally.

This PR is a follow up of #768 , should be ready when it's merged.

This PR completely removes SelectResult from GRPC returned result. Now GRPC's do_query(select) returns FlightData.

Some prometheus remote read protocol transformation (formly from SelectResult, now from RecordBatch) required detailed review from @killme2008 , PTAL.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: tests/runner/src/env.rs [hunk 1]
```diff
@@ -18,11 +18,8 @@ use std::process::Stdio;
 use std::time::Duration;
 
 use async_trait::async_trait;
-use client::api::v1::codec::SelectResult;
-use client::api::v1::column::SemanticType;
-use client::api::v1::ColumnDataType;
 use client::{Client, Database as DB, Error as ClientError, ObjectResult, Select};
-use comfy_table::{Cell, Table};
```
Comment (waynexia) [inline tests/runner/src/env.rs:25]: Maybe this can be removed from Cargo.toml

File: tests/runner/src/env.rs [hunk 2]
```diff
@@ -129,75 +126,20 @@ impl Display for ResultDisplayer {
...
                 }
-                // TODO(LFC): Implement it.
-                ObjectResult::FlightData(_) => unimplemented!(),
+                ObjectResult::FlightData(messages) => {
+                    let pretty = flight::flight_messages_to_recordbatches(messages.clone())
+                        .map_err(|e| e.to_string())
+                        .and_then(|x| x.pretty_print().map_err(|e| e.to_string()));
+                    match pretty {
+                        Ok(s) => write!(f, "{s}"),
+                        Err(e) => write!(f, "err: {e}"),
```
Comment (waynexia) [inline tests/runner/src/env.rs:138]:
```suggestion
                        Err(e) => write!(f, "format result error: {e}"),
```
Prefer to specify this error comes from formatting

File: src/common/grpc/src/select.rs [hunk 3]
```diff
@@ -12,107 +12,18 @@
...
-        };
-        columns.push(column);
-    }
-
-    Ok(SelectResult {
-        columns,
-        row_count: row_count as u32,
-    })
-}
+use crate::error::{ConversionSnafu, Result};
```
Comment (waynexia) [inline src/common/grpc/src/select.rs:26]: Looks like functions in this file are only used in tests. I'm wondering if we can annotate this mod with `#[cfg(test)]`

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by MichaelScofield** [comment]
@waynexia PTAL

**Review by waynexia — APPROVED** [review]
LGTM

**Pull request merged**